pkg/apis/baremetal/v1alpha1/baremetalmachineproviderspec_types.go

```diff
@@ -73,14 +76,22 @@ type Image struct {
 	Checksum string `json:"checksum"`
 }
 
+// Custom deploy is a description of a customized deploy process.
+type CustomDeploy struct {
+	// Custom deploy method name.
+	// This name is specific to the deploy ramdisk used. If you don't have
+	// a custom deploy ramdisk, you shouldn't use CustomDeploy.
+	Method string `json:"method"`
+}
+
 // IsValid returns an error if the object is not valid, otherwise nil. The
 // string representation of the error is suitable for human consumption.
 func (s *BareMetalMachineProviderSpec) IsValid() error {
 	missing := []string{}
-	if s.Image.URL == "" {
+	if s.CustomDeploy.Method == "" && s.Image.URL == "" {
 		missing = append(missing, "Image.URL")
 	}
-	if s.Image.Checksum == "" {
+	if s.CustomDeploy.Method == "" && s.Image.Checksum == "" {
```

Comment:
Arguably we should always check for a checksum when there is an image, not just if there is no custom deploy method. But for now there's no difference.